pkg/controllers/machine_controller.go

```diff
@@ -104,6 +108,7 @@ type MachineReconciler struct {
 	libvirt           *libvirt.Libvirt
 	guestCapabilities guest.Capabilities
 	tcMallocLibPath   string
+	DomainTuners      []TuneFunc
```

Comment:
@lukasfrank is reconciler running in multiple goroutines in libvirt-provider? Because virtlet runs with 1 worker in time. This concept can create unexpected behavior with multiple goroutines.

Reply:
we discussed it on sync call. libvirt-provider can process 10 machine at the same time and call tune functions in the same time. It can create unexpected behavior because each tuna function calculate resources independently.

Reply:
So, is it like we need to make the resource calculation thread safe ?

Reply:
it depends how do you mean it but technically yes. Primary problem is missing central management for pinning vcpu.

Reply:
Central management will be taken care under the the Resource manager umbrella I believe. So, what kind of modifications are you expecting in this PR.

Reply:
i am not sure about it. pinning cpu isn't resource. It is some state, how vms are assign to cpu cores, what isn't important during scheduling. But it is for long discussion.
But minimally, if we don't want to improve code in this PR, it will ideally change worker count to 1 for minimize impact of this problem.